w07h03/FieldVisualizeHelper.java

```diff
@@ -0,0 +1,113 @@
+     * </p><p><b>Example field in FDL</b>
+     * <pre><code>
+     *     . g g h . h w
+     *     w p . . . . .
+     *     . . . . . . .
+     * </code></pre>
+     * Alternatively:
+     * <pre><code>
+     *     . g g h . h w w p . . . . . . . . . . . .
+     * </code></pre></p>
+     *
+     * @param cellsTextual A String/text block containing the field description in FDL
+     * @param width        The width of the field
+     * @param height       The height of the field
+     * @return A Cell-Array with the entities created
+     */
+    public static Cell[] fieldDescriptionToCellArray(String cellsTextual, int width, int height) {
+        cellsTextual = cellsTextual.replace("\n", " ");
+        String[] cellFieldTextual = cellsTextual.split(" ");
+        Cell[] cellField = new Cell[width * height];
+
+        for (int i = 0; i < width * height; i++) {
+            cellField[i] = switch (cellFieldTextual[i]) {
+                case "g" -> new Plant();
+                case "h" -> new Hamster();
+                case "w" -> new Wolf();
+                case "p" -> new Pingu();
+                default -> null;
+            };
+        }
+        return cellField;
+    }
+
+    /**
+     * Turns a Cell-Array into a field description.
+     * <p>
+     * <b>Field Description Language (FDL)</b><br>
+     * FDL uses the following syntax:
+     * <ul>
+     *     <li><code>g/🌱</code> = Plant ("Greenery")</li>
+     *     <li><code>h/🐹</code> = Hamster</li>
+     *     <li><code>p/🐧</code> = Pingu</li>
+     *     <li><code>w/🐺</code> = Wolf</li>
+     *     <li><code>./🔳</code> = Empty Cell</li>
+     * </ul><br>
+     * Cells are separated by spaces, lines will be, for better readability, be separated by line breaks.<br>
+     * </p><p><b>Example field in FDL</b>
+     * <pre><code>
+     *     . g g h . h w
+     *     w p . . . . .
+     *     . . . . . . .
+     * </code></pre>
+     * Alternatively:
+     * <pre><code>
+     *     🔳 🌱 🌱 🐹 🔳 🐹 🐺
+     *     🐺 🐧 🔳 🔳 🔳 🔳 🔳
+     *     🔳 🔳 🔳 🔳 🔳 🔳 🔳
+     * </code></pre></p>
+     *
+     * @param cellField A Cell-Array to be turned into an FDL description
+     * @param width     The width of the field
+     * @param height    The height of the field
+     * @param useEmojis Specifies whether the output should be given as emojis representing the respective cells. Might make it easier to get an overview of the field.
+     * @return A String with the textual description of cellField in FDL
+     */
+    public static String cellArrayToFieldDescription(Cell[] cellField, int width, int height, boolean useEmojis) {
+        String[] charSetToUse = useEmojis ? emojiSymbols : textualSymbols;
+
+        StringBuilder sb = new StringBuilder();
+        for (int y = 0; y < height; y++) {
+            for (int x = 0; x < width; x++) {
+                Cell c = cellField[y * width + x];
+                if (c instanceof Plant) {
```

Comment:
I guess you could simplify that code by using a hash map that maps each class to an index.

Reply:
Both ways would be cleaner, but break the emoji functionality. Switch requires constant values, so we would have to hardcode chars. We could use two maps, but then need another if statement to choose the right one. Easiest to just leave it as it is.